File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> cmp;

        for (int i = 0; i < nums.size(); i ++) {
            int ele = nums[i];
            

            if (cmp.count(target - ele) == 1) {
                return {cmp[target - ele], i};
            }

            cmp[ele] = i;

        }
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        is_duplicate = False
        all_values = set()
        for x in nums:
            if x in all_values:
                is_duplicate =  True
            else:
                all_values.add(x)
        
        return is_duplicate
         